components/widgets/header/header.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, ReactNode, Fragment, HTMLAttributes} from 'react';
+import classNames from 'classnames';
+
+import './header.scss';
+
+type Props = {
+    heading: ReactNode;
+    level?: 0 | 1 | 2 | 3 | 4 | 5 | 6;
```

Comment:
this should be a  named type, I'm guessing this could be reused in future components

Reply:
I'm a huge fan of reusability, but I think `level` is fairly specific to `Headings` below, though it's still accessible via `React.ComponentProps<typeof Header>['level']` if need be, or I could export the props type?